Simplenote/Classes/OptionsViewController.swift

```diff
@@ -372,7 +372,23 @@ private extension OptionsViewController {
     @IBAction
     func publishWasPressed(_ newState: Bool) {
         SPTracker.trackEditorNotePublishEnabled(newState)
-        SPObjectManager.shared().updatePublishedState(newState, note: note)
+        let publishController = SPAppDelegate.shared().publishController
+        publishController.updatePublishState(for: note, to: newState, completion: { (state) in
+            switch state {
+            case .publishing:
+                let notice = NoticeFactory.publishing
+                NoticeController.shared.present(notice)
+            case .published:
+                let notice = NoticeFactory.published(self.note)
+                NoticeController.shared.present(notice)
+            case .unpublishing:
+                let notice = NoticeFactory.unpublishing
+                NoticeController.shared.present(notice)
+            case .unpublished:
+                let notice = NoticeFactory.unpublished(self.note)
+                NoticeController.shared.present(notice)
```

Comment:
(No need to change anything!!!)

Another alternative to this, is to use an extension on Notice (in a standalone file), and do this:

```
extension Notice {
    static func unpublishingNotice() { ... }
    static func unpublishedNotice(note: Note) { .. }
}
```

That enables this syntax:

```
NoticeController.shared.present( .unpublishingNotice() )
```

Both approaches are 100% correct, no need to change anything!

Reply:
I thought about doing this a while back, and decided to go with the factory so that the notices were not coupled to Simplenote. I see how doing this in an extension still keeps the simplenote specific details of the notices being used private to simplenote 🤔  I think for now I like it as is, but would be open to changing to the using an extension instead